src/envoy/http/service_control/client_cache.cc

```diff
@@ -336,6 +337,20 @@ void ClientCache::handleCheckResponse(const Status& http_status,
             *response, config_.service_name(), &response_info);
     // Check errors should be displayed to the client.
     translate_non_5xx = false;
+
+    switch (response_info.error_type) {
+      case CheckResponseErrorType::CONSUMER_BLOCKED:
+        filter_stats_.filter_.denied_consumer_blocked_.inc();
+        break;
+      case CheckResponseErrorType::CONSUMER_ERROR:
+      case CheckResponseErrorType::SERVICE_NOT_ACTIVATED:
+      case CheckResponseErrorType::API_KEY_INVALID:
+        filter_stats_.filter_.denied_consumer_error_.inc();
+        break;
+      default:
+        break;
```

Comment:
This function is long enough so put this part into a function?

Reply:
I think it's fine, it's a simple switch statement. I would like all the SC Check related stats incremented in one function (see a few lines below, where we increment stats for 5xx). Otherwise, it's difficult to reason about which code paths changed which stats.